pwa/src/templates/componentDetail/ComponentsDetailTemplate.module.css

```diff
@@ -45,8 +45,16 @@
   flex: 2;
 }
 
+.logoContainer {
+  max-height: var(--web-app-component-detail-logo-height);
+  max-width: var(--web-app-component-detail-logo-width);
+  text-align: center;
+}
+
 .componentImage {
   max-width: var(--web-app-component-image-max-width);
+  max-height: 100%;
+  object-fit: contain;
 }
```

Comment:
Whenever setting the height and width of an image, using a container is a good idea. However, the container should be in control of the width and height of the image afterwards, which is not (really) the case here.

```
.logoContainer {
  width: var(--web-app-component-detail-logo-width); // not max-width
  height: var(--web-app-component-detail-logo-height); // not max-height
  // note: its children will always be 100% filling, so text-align is redundant.
}

.componentLogo {
  width: 100%; // always fill its container
  height: 100%; // always fill its container
  object-fit: cover;
}
```

Besides above refactoring, I'd suggest renaming `.componentImage` to either `.componentLogo` or just `.logo`. Since it's the child of the `logoContainer`.

Reply:
There is, however, the need to set a max-width on the `container`. I'd suggest setting this to a fixed value _without_ the use of a design token (it's a must-have fixed value, which should not be dynamically changed by the theme).